lib/app/modules/settings/controllers/theme_controller.dart

```diff
@@ -1,32 +1,99 @@
+import 'dart:async';
 import 'package:flutter/material.dart';
 import 'package:get/get.dart';
 import 'package:ultimate_alarm_clock/app/data/providers/secure_storage_provider.dart';
 import 'package:ultimate_alarm_clock/app/utils/constants.dart';
+import 'dart:developer' as dev;
 
 class ThemeController extends GetxController {
   var isLightMode = true.obs;
   final _secureStorageProvider = SecureStorageProvider();
 
+  late Timer _brightnessCheckTimer;
+  bool isSystemModeActive = false;
```

Comment:
You can just check this from the theme details that we're alreading storing, this variable would not be needed

Reply:
I tried that, but the problem was that, either the system mode text won't update, or else it won't change completely to light or dark issue. Weird issue but this was the most optimized way I found out

Reply:
Could you elaborate more on this? We are storing the theme details in storage and fetching it in `onInit` so we should have access to the option chosen by the user, yes?

Reply:
In this existing method, we can retrieve the current theme settings. But it isn't working properly for the system mode. As in, all the elements/widgets are not changing accordingly. So I had to do it manually itself. For that I created a new variable. Again it is a weird issue, but this was the only work around for it. Its the optimal one for now.